kvclientpool.go

```diff
@@ -202,9 +229,11 @@ func (p *kvClientPool) startNewClientLocked() <-chan struct{} {
 		}
 
 		if err != nil {
-			p.logger.Warn("failed to create a new client connection", zap.Error(err))
-
-			p.connectErr = err
+			// Don't store the error if we've attempted to write an op after closing a connection,
+			// just recreate the client if required.
+			if !errors.Is(err, net.ErrClosed) {
```

Comment:
This seems dubious.  I don't think we should be special-casing errors here.  Any error which leads the client to being unavailable should count an error from the perspective of being able to tell the user why their operations can't execute.

Reply:
This wasn't actually supposed to make it this far, I was using this check to debug something.